### PR TITLE
Fix: Dismiss lingering tooltips when the page is scrolled

### LINKED ISSUE #3486: [Dashboard] Tooltips display incorrect behavior on Dashboard screen on mobile
## Steps to reproduce

1. Open Dashboart
2. Tap on the "Copy contact address"  and "Copy this colony URL" icons
3. Scroll page up and down several times

## Expected behaviour

* Tooltips should disappear after scrolling up or down

## Actual behaviour

* Tooltips remain visible even after scrolling

https://github.com/user-attachments/assets/df17ba38-158d-45ed-97cd-2cf6de484f29

### PR DESCRIPTION
**Pull request by rumzledz**
## Description

TLDR: The `trigger` and `isOpen` props were clashing and led to situations whereby the tooltip still lingers even if we don't want it to. As a workaround, I had to manually detect hover states for the Tooltip's children and conditionally update the `isOpen` prop ✌️ 

| Production | Current Master | This Branch |
| ---- | ---- | --- |
| Production | master | my branch |

This is how it behaves on mobile now, on this branch:

mobile-tooltip

## Testing

1. Hover on the Copy Contract Address button
2. Click it
3. Scroll the page
4. Verify that its tooltip is not visible
5. Do the same for the Share button please

## Diffs

**New stuff** ✨

A new hook called `useOnElementScroll` which executes a function when the page is scrolled:

```js
useOnElementScroll({
  scrollableElementId: 'scrollable-element',
  callback: () => console.log('Scrolled!'),
});
```

I also had to slightly tweak the `useCopyToClipboard` hook.

Resolves #3486

### CODE REVIEW

**Review by rdig — APPROVED** [review]
Tooltip is no longer rendered when scrolling. Very nice fix here!


https://github.com/user-attachments/assets/9caf3c38-dfc1-46b8-a93a-77275f76fb2c


https://github.com/user-attachments/assets/28cad51a-41c4-48a1-b3e3-535b5afb9895

File: src/hooks/useOnElementScroll.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    const scrollElement = document.getElementById(scrollableElementId);
+
+    if (!scrollElement) {
+      console.error(`Element with ID '${scrollableElementId}' not found`);
+      return noop;
+    }
+
+    const handleScroll = () => callback();
+
+    scrollElement.addEventListener('scroll', handleScroll, { passive: true });
```
Comment (iamsamgibbs) [inline src/hooks/useOnElementScroll.tsx:41]: Can we throttle handleScroll here?

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Nice job, this is working as expected. Disappears on scroll. Still disappears after expected amount of time.

On desktop:
https://github.com/user-attachments/assets/ab9044c2-8ea2-47fd-83fe-8b02da79a5f9

On mobile:
https://github.com/user-attachments/assets/f3a42757-c329-403e-af7f-81891939337b

I'm just rejecting as this tooltip has the wrong text when copied (I've checked master and it isn't an issue there, so is something introduced by this PR):

Screenshot 2024-11-07 at 13 41 29

Also left a comment about throttling the scroll event listener.

File: src/hooks/useOnElementScroll.tsx (hunk 1 above)
Comment (rumzledz) [reply src/hooks/useOnElementScroll.tsx:41]: Can do! And nice spot with the copy, that's my bad!

Comment (rumzledz) [reply src/hooks/useOnElementScroll.tsx]: All right added a `scrollThrottle` prop set to 100ms by default and updated the copy! 🚀

**Review by iamsamgibbs — APPROVED** [review]
Thanks for fixing the tooltip and implementing the throttle!

Screenshot 2024-11-07 at 16 09 05

**Review by Nortsova — APPROVED** [review]
Very nice solution! 🙌 

I investigated tooltip documentation and it should be something with a boundary, but unfortunately no luck to implement that "boundary" thing. So I am happy that you found custom solution 🥳  

tooltip-disapear

**Review by rdig — APPROVED** [review]
